packages/console/src/pages/ApplicationDetails/index.tsx

```diff
@@ -1,46 +1,149 @@
 import { Application } from '@logto/schemas';
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
-import { useParams } from 'react-router-dom';
+import { useLocation, useParams } from 'react-router-dom';
 import useSWR from 'swr';
 
 import BackLink from '@/components/BackLink';
 import Button from '@/components/Button';
 import Card from '@/components/Card';
 import CopyToClipboard from '@/components/CopyToClipboard';
+import FormField from '@/components/FormField';
 import ImagePlaceholder from '@/components/ImagePlaceholder';
+import TabNav, { TabNavLink } from '@/components/TabNav';
+import TextInput from '@/components/TextInput';
 import { RequestError } from '@/swr';
 import { applicationTypeI18nKey } from '@/types/applications';
 
 import * as styles from './index.module.scss';
 
+type OidcConfig = {
+  authorization_endpoint: string;
+  userinfo_endpoint: string;
+  token_endpoint: string;
+};
+
 const ApplicationDetails = () => {
   const { id } = useParams();
+  const location = useLocation();
+
   const { data, error } = useSWR<Application, RequestError>(id && `/api/applications/${id}`);
+  const { data: oidcConfig, error: fetchOidcConfigError } = useSWR<OidcConfig, RequestError>(
+    '/oidc/.well-known/openid-configuration'
+  );
+  const isLoading = !data && !error && !fetchOidcConfigError;
+  const dataFetched = data && oidcConfig;
+
+  const { handleSubmit, register, reset } = useForm<Application>({
+    defaultValues: data,
+  });
+
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
-  const isLoading = !data && !error;
+
+  const isNormalSettings = !location.pathname.includes('advanced-settings');
+
+  const buildTabLink = (tabLinkName: string) => {
+    const index = location.pathname.lastIndexOf('/');
+
+    if (index < 0) {
+      return location.pathname;
+    }
+
+    return `${location.pathname.slice(0, Math.max(0, index))}/${tabLinkName}`;
```

Comment:
If you want to remove the last `/`, you can use
```typescript
.slice(0, -1)
```

Reply:
I just want to replace the last path segment with the correct tab name.
We have a path: `https://www.logto.dev/console/applications/some-id/settings`.
What I want to do is remove the last `/settings` and change it to `/advanced-settings`.
Finally, We get `https://www.logto.dev/console/applications/some-id/advanced-settings`.

Reply:
updated